core/player/src/view/parser/utils.ts

```diff
@@ -1,8 +1,28 @@
 import type { Node } from "./types";
 
-/** Check to see if the object contains async */
-export function hasAsync(obj: object): boolean {
-  return Object.prototype.hasOwnProperty.call(obj, "async");
+/**
+ * Checks if there are templated values in the object
+ *
+ * @param obj - The Parsed Object to check to see if we have a template array type for
+ * @param localKey - The key being checked
+ */
+export function hasTemplateValues(obj: any, localKey: string) {
+  return (
+    Object.hasOwnProperty.call(obj, "template") &&
+    Array.isArray(obj?.template) &&
+    obj.template.length &&
+    obj.template.find((tmpl: any) => tmpl.output === localKey)
+  );
+}
+
+/** Check to see if the string is a valid switch key */
+export function hasSwitchKey(localKey: string) {
```

Comment:
This is not currently used outside of `switch.ts`. The only reason I have it here in `utils.ts` is to keep it consistent with `hasTemplateKey`. I could move it to `switch.ts` though if you don't think there's value in keeping it here.

Reply:
I don't have strong opinions on this, I pointed it out because I thought this was similar to the functions in each plugin that checks the type (replacing `determineNodeType`) as they're only relevant to said plugin. You can decide what to do here